src/App.tsx

```diff
@@ -21,75 +22,125 @@ enum SortType {
+  });
+
+  if (isReversed) {
+    return visibleGoods.reverse();
+  }
 
   return visibleGoods;
 }
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+type State = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+export class App extends React.Component<{}, State> {
+  state: State = {
+    sortType: SortType.NONE,
+    isReversed: false,
+  };
+
+  reverse = () => {
+    this.setState(state => ({
+      isReversed: !state.isReversed,
+    }));
+  };
+
+  alphabetSort = () => {
+    return this.setState({ sortType: SortType.ALPHABET });
+  };
+
+  lengthSort = () => {
+    this.setState({ sortType: SortType.LENGTH });
+  };
+
+  reset = () => {
+    this.setState({
+      sortType: SortType.NONE,
+      isReversed: false,
+    });
+  };
+
+  render() {
+    const sortedList = getReorderedGoods(
+      goodsFromServer,
+      this.state,
+    );
+
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            className={classNames('button is-info', {
+              'is-light': this.state.sortType !== SortType.ALPHABET,
+            })}
+            onClick={this.alphabetSort}
+          >
+            Sort alphabetically
+          </button>
+
+          <button
+            type="button"
+            className={classNames('button is-success', {
+              'is-light': this.state.sortType !== SortType.LENGTH,
+            })}
+            onClick={this.lengthSort}
+          >
+            Sort by length
+          </button>
+
+          <button
+            type="button"
+            className={classNames('button is-warning', {
+              'is-light': this.state.isReversed !== true,
+            })}
+            onClick={this.reverse}
+          >
+            Reverse
+          </button>
+
+          {
+            this.state.sortType !== SortType.NONE
+            || this.state.isReversed === true
+              ? (
+                <button
+                  type="button"
+                  className="button is-danger is-light"
+                  onClick={this.reset}
+                >
+                  Reset
+                </button>
+              )
+              : true
```

Comment:
let's use different type of conditional rendering.
Hint: `{someValue && <SomeComponent>}`